api/simple_authentication.py

```diff
@@ -31,9 +31,16 @@ class SimpleAuthenticationProvider(BasicAuthenticationProvider):
     TEST_NEIGHBORHOOD = 'neighborhood'
 
     basic_settings = list(BasicAuthenticationProvider.SETTINGS)
-    for setting in basic_settings:
-        if setting['key'] == BasicAuthenticationProvider.TEST_PASSWORD:
-            setting['required'] = True
```

Comment:
This line is changing a value in a dict referenced in `BaseAuthenticationProvider.SETTINGS`, so was have a broader than intended effect.